File: BOJ/1110.c
// 1110 더하기 사이클
#include <stdio.h>

int solution(int n) {
    int answer = 0;

    int number = n, new_number = 0;

    while (1) {
        int front = number / 10;
        int back = number % 10;

        new_number = back * 10 + ((front + back) % 10);
        answer++;

        if (new_number == n)
            break;
        
        number = new_number;
    }
    return answer;
}

int main(void) {
    int n;

    scanf("%d", &n);

    int ret = solution(n);
    printf("%d", ret);

    return 0;
}

File: BOJ/1110.py
# 1110 더하기 사이클
n = int(input())

number = n
answer = 0

while True:
    front = number // 10
    back = number % 10
    
    number = back * 10 + (front + back) % 10
    answer += 1
    
    if n == number:
        break

print(answer)